Lib/test/test_typing.py

```diff
@@ -7769,6 +7769,46 @@ class Cat(Animal):
             'voice': str,
         })
 
+    def test_keys_inheritance_with_same_name(self):
+        class NotTotal(TypedDict, total=False):
+            a: int
+
+        class Total(NotTotal):
+            a: int
+
+        self.assertEqual(NotTotal.__required_keys__, frozenset())
+        self.assertEqual(NotTotal.__optional_keys__, frozenset(['a']))
+        self.assertEqual(Total.__required_keys__, frozenset(['a']))
+        self.assertEqual(Total.__optional_keys__, frozenset())
+
+        class Base(TypedDict):
+            a: NotRequired[int]
+            b: Required[int]
+
+        class Child(Base):
+            a: Required[int]
+            b: NotRequired[int]
+
+        self.assertEqual(Base.__required_keys__, frozenset(['b']))
+        self.assertEqual(Base.__optional_keys__, frozenset(['a']))
+        self.assertEqual(Child.__required_keys__, frozenset(['a']))
+        self.assertEqual(Child.__optional_keys__, frozenset(['b']))
+
+    def test_multiple_inheritance_with_same_key(self):
+        class Base1(TypedDict):
+            a: NotRequired[int]
+
+        class Base2(TypedDict):
+            a: Required[str]
+
+        class Child(Base1, Base2):
+            pass
+
+        # Last base wins
```

Comment:
Is it correct that the last base should win? Doesn't that go against how multiple inheritance in Python usually works, where earlier bases in the `__bases__` tuple generally have priority:

```pycon
Python 3.13.0a2+ (heads/main:e9d1360c9a, Nov 24 2023, 11:23:45) [MSC v.1932 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> class Foo:
...     x = 1
...
>>> class Bar:
...     x = 2
...
>>> class Baz(Foo, Bar): pass
...
>>> Baz.x
1
```

Reply:
It's arguably not correct as you say, but it's the current behavior for `__annotations__` and changing that behavior seems difficult.

```
>>> class A(TypedDict):
...     a: int
... 
>>> class B(TypedDict):
...     a: str
... 
>>> class C(A, B): pass
... 
>>> C.__annotations__
{'a': <class 'str'>}
```

Reply:
Oof, that seems unfortunate. As you say, though, better to be internally consistent for now, I guess!